### ISSUE TITLE
Misuses of cryptographic APIs

### ISSUE THREAD

**Issue by mhp0rtal** [DigitalPebble/storm-crawler#736]
Hi
The following lines have cryptographic API misuses. <br> File name => external/aws/src/main/java/com/digitalpebble/stormcrawler/aws/bolt/CloudSearchUtils.java: Line number => 36: API name => MessageDigest:Operation on object of type java.security.MessageDigest object not completed. Expected call to digest, update<BR>

**Comment by jnioche** [comment]
thanks, do you want to provide a pull request for this?

**Comment by rzo1** [comment]
I dont think, that this is a real issue? `MessageDigest` is used for generating a normalised doc ID based on the URL of the given document and not as a cryptographic hash function, so this is not a security issue.

**Comment by mhp0rtal** [comment]
Hi,
Thanks for your attention.
I reported the cryptographic misuses due to my recent research on how developers employ Java Cryptography Architecture (JCA) APIs in Java applications. I utilized some static analysis tools intended for checking cryptographic practices. The aforementioned misuses in those files were obtained from the results of the tools. The severity of the bug is relative. For instance, Even though MD5 is listed on Java documentation, MD5 is not recommended to use in any language anymore as it is considered insecure. An attacker can try billions of candidate passwords per second on a single GPU. Nevertheless, depending on the scenario the same hash algorithm may not be a problem for file integrity purpose.
There are several instances of cryptographic misuses which could lead to a security vulnerability:
*Do not use ECB mode for encryption.
*Do not use a non-random IV for CBC encryption
*Do not use constant encryption keys.
*Do not use constant salts for PBE.
*Do not use fewer than 1,000 iterations for PBE.
*Do not use static seeds to seed SecureRandom(·)
Well, your justification can be true as we did not dig into the context of the API usage
